src/Data/Functor/Bind/Class.hs

```diff
@@ -279,6 +280,9 @@ instance Apply Complex where
   (a :+ b) <.> (c :+ d) = a c :+ b d
 #endif
 
+instance Apply Q where
+  (<.>) = (<*>)
```

Comment:
`Q` has only been an `Applicative` instance since `template-haskell-2.7.0.0`, so this won't compile on GHC 7.0 and 7.2. You should be able to work around this with CPP, though:

```haskell
instance Apply Q where
#if MIN_VERSION_template_haskell(2,7,0)
  (<.>) = (<*>)
#else
  (<.>) = ap
#endif
```